src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunctionSet.cs

```diff
@@ -0,0 +1,273 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerFx.Core.Utils;
+
+namespace Microsoft.PowerFx.Core.Functions
+{
+    internal class TexlFunctionSet<T>
+        where T : ITexlFunction
+    {
+        private Dictionary<string, HashSet<T>> _functions;
+        private Dictionary<string, HashSet<T>> _functionsInvariant;
+        private int _count;
+
+        internal TexlFunctionSet()
+        {
+            _functions = new Dictionary<string, HashSet<T>>();
+            _functionsInvariant = new Dictionary<string, HashSet<T>>();
+            _count = 0;
+        }
+
+        internal TexlFunctionSet(T function)
+            : this()
+        {
+            if (function == null)
+            {
+                throw new ArgumentNullException($"{nameof(function)} cannot be null", nameof(function));
+            }
+
+            _functions.Add(function.Name, new HashSet<T>() { function });
+            _functionsInvariant.Add(function.LocaleInvariantName, new HashSet<T>() { function });
```

Comment:
>LocaleInvariantName [](http://example.com/codeflow?start=45&length=19)

I think, although not 1000% sure, that LocaleInvariantName == Name and we never have locale-specific function names.

Reply:
I agree but wanted to be 100% compatibile even if this part is never used.
If LocalInvariantName is never used, let's remove it and we'll save CPU & memory

Reply:
Can be considered in a different PR if we need